heim-process/src/sys/windows/bindings/token.rs

```diff
@@ -0,0 +1,66 @@
+use std::ptr;
+use winapi::shared::minwindef::{DWORD, LPVOID};
+use winapi::um::handleapi::CloseHandle;
+use winapi::um::processthreadsapi::OpenProcessToken;
+use winapi::um::securitybaseapi::GetTokenInformation;
+use winapi::um::winnt::{TokenUser, HANDLE, TOKEN_QUERY, TOKEN_USER};
+
+use heim_common::prelude::*;
+use heim_common::sys::windows::Handle;
+use heim_common::Result;
+use heim_host::os::windows::UserExt;
+use heim_host::User;
+
+pub struct Token(HANDLE);
+
+impl Token {
+    pub fn open(process_handle: &Handle) -> Result<Self> {
+        let mut token_handle: HANDLE = ptr::null_mut();
+
+        let result = unsafe { OpenProcessToken(**process_handle, TOKEN_QUERY, &mut token_handle) };
+
+        if result == 0 {
+            return Err(Error::last_os_error().with_ffi("OpenProcessToken"));
+        }
+
+        Ok(Self(token_handle))
+    }
+
+    pub fn user(&self) -> Result<User> {
+        // data should always be 44 bytes
+        let mut data: Vec<u8> = Vec::with_capacity(64);
```

Comment:
I feel like `MaybeUninit` should be used in here instead, so maybe it is worth to investigate why it was failing before, as we discussed.